auth/service.hh

```diff
@@ -75,6 +75,7 @@ public:
 /// internal data on which auth::service depends.
 struct command_desc {
     const resource& resource; ///< Resource impacted by this command.
+    bool alter_with_opts; ///< True iff command is ALTER ... WITH ...
```

Comment:
This should be an enum (and alter_with_opts one of the choices, and `other` as another). If we add `drop` as the next option, only one of `drop` or `alter_with_opts` can hold.

Perhaps it should be a variant, so each choice can carry some extra data if needed (that can be done if/when needed, but I think it's the logical data structure).

Reply:
Made it enum.